### DISCUSSION TITLE
Foundry Fund Me: How does forge snapshot work?

### DISCUSSION THREAD

**Discussion by spencerj411** [Cyfrin/foundry-full-course-cu#2189] [Q&A]
Hi, I'm currently watching how to [Optimise the withdraw function gas costs](https://updraft.cyfrin.io/courses/foundry/foundry-fund-me/optimise-solidity-function-gas-costs?lesson_format=video). For fun, I implemented the tests slightly differently to how it was done in the video. I noticed that when I run `gas snapshot` I'm getting lower gas. For example, **206033** for `testWithdrawWithMultipleFunders() `instead of **487915** in the video. 

Does forge snapshot only take into account the gas used to interact with the contract that you are trying to test or does it consider the whole test function? If I were to implement forge snapshot, it should only be the former. I find it weird that the numbers are different. If latter, how come it was implemented this way? Could someone please help me understand? Thanks in advance!

**Comment by EngrPips** [comment]
Hello @spencerj411, I haven't read the Foundry docs yet, and I am not sure how forge snapshot works, but here is the link https://book.getfoundry.sh/forge/gas-snapshots to the part of the docs that talked about gas snapshot. You may find your answer there.
